### PR TITLE
Initial worflow for `sourmash` method

### PR DESCRIPTION
**Pull request by kiepczi**
This PR provides:

- Initial workflow for `sourmash` method. The workflow consists of two rules: `sketch`, which creates DNA signatures, and `compare`, which estimates ANI values with parameters like `--max-containment` and `--estimate-ani`. Currently, there are no default settings for the `k` and `scaled` parameters in the `sketch` rule. As discussed last week, we will likely want to set these to the `sourmash` defaults (`k=31, scaled=1000`); however, since these settings fail to estimate ANI values for the current viral test set, I have left them undefined, and they must be provided to run the workflow.
- Generated fixtures such as `.sig` and `.csv` files for testing purposes. These files were generated using `k=31, scaled=300`. The parameters were chosen without any biologically meaningful reasons, but they can estimate ANI and do not produce large `.sig` files, which would otherwise not be added to the repository due to their size.
- `sourmash` was added to `requirements.txt` and `tools.py`.
- The workflow was tested with `pytest` to ensure that it runs smoothly and produces the expected files.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [X] This change requires a documentation update
- [ ] This is a documentation update

## Action Checklist

- [X] Work on a single issue/concept (if there are multiple separate issues to address, please use a separate pull request for each)
- [X] Fork the `pyani-plus` repository under your own account (please [allow write access for repository maintainers](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
- [ ] Set up an appropriate development environment (please see `CONTRIBUTING.md` for this repository)
- [X] Create a new branch with a short, descriptive name
- [X] Work on this branch
  - [X] style guidelines have been followed (please see `CONTRIBUTING.md` for this repository)
  - [X] new code is commented, especially in hard-to-understand areas
  - [ ] corresponding changes to documentation have been made
  - [X] tests for the change have been added that demonstrate the fix or feature works
- [X] Test locally with `pytest-plus -v` **non-passing code will not be merged**
- [ ] Rebase against `origin/master`
- [X] Check changes with linting/formatting tools before submission (please see `CONTRIBUTING.md` for this repository)
- [X] Commit branch
- [X] Submit pull request, describing the content and intent of the pull request
- [X] Request a code review
- [X] Continue the discussion at [`Pull requests` section](https://github.com/widdowquinn/pyani-plus/pulls) in the `pyan-plus` repository

### CODE REVIEW

**Comment by kiepczi** [comment]
This that we need to discuss:
- The estimation of ANI values using the `--max-containment` parameter is symmetrical (e.g., genome A vs. genome B returns the same value as genome B vs. genome A). `--containment` is, however, not symmetrical and returns slightly different values depending on the order of the compared genomes. If we choose to use only `--max-containment` as the default setting without providing end users an option, there is no need to run comparisons in both directions. However, if we decide to give end users that option, we will need to run comparisons in both directions to be consistent with other methods like `aniM` and `aniB`. The question is: what do we want to do?

- It also appears that `sourmash` always returns an ANI of 1.0% (on a scale of 0-1%) for self-to-self comparisons. Do we want `pyANI` plus to still run self-to-self comparisons?

**Comment by widdowquinn** [comment]
>     * The estimation of ANI values using the `--max-containment` parameter is symmetrical (e.g., genome A vs. genome B returns the same value as genome B vs. genome A). `--containment` is, however, not symmetrical and returns slightly different values depending on the order of the compared genomes. If we choose to use only `--max-containment` as the default setting without providing end users an option, there is no need to run comparisons in both directions. However, if we decide to give end users that option, we will need to run comparisons in both directions to be consistent with other methods like `aniM` and `aniB`. The question is: what do we want to do?

Let's use `--max-containment` as the default, but make other options, like `--containment` available to users. Initially let's just run all-vs-all in both directions and, if we need to optimise later, we can run query_vs_subject and populate two cells in the results matrix for `--max-containment` only. This ties in with @peterjc suggestion of a `--mode` option in #165 

>     * It also appears that `sourmash` always returns an ANI of 1.0% (on a scale of 0-1%) for self-to-self comparisons. Do we want `pyANI` plus to still run self-to-self comparisons?

It is inevitable that `sourmash` should return 1.0 for self-vs-self. But maybe sometimes it doesn't work right ;) (I'm sure it always works correctly, but it's better to be honest, rather than guess - trust but verify).

**Review by widdowquinn — CHANGES_REQUESTED** [review]
Great progress - I'm excited to see the sourmash vs ANIm vs ANIb comparisons. I think we need to change the JSON comparisons to just ignore the `filename` field, rather than modify to the extent that's here, mabe?

File: tests/snakemake/test_snakemake_sourmash_workflow.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    """
+    # Remove the output directory to force re-running the snakemake rule
+    shutil.rmtree(sourmash_targets_signature_outdir, ignore_errors=True)
+
+    config = config_sourmash_args.copy()
+    config["outdir"] = sourmash_targets_signature_outdir
+
+    # Run snakemake wrapper
+    runner = SnakemakeRunner("snakemake_sourmash.smk")
+    runner.run_workflow(sourmash_targets_sig, config, workdir=Path(tmp_path))
```
[the remark this answers is not in the document]
Comment (kiepczi) [reply tests/snakemake/test_snakemake_sourmash_workflow.py]: I was chatting to Hosski and didn’t realise, this change was never pushed to remote branch. That should now be fixed.

**Review by peterjc — APPROVED** [review]
This looks good - with logging to the DB via the private CLI the next step, followed by adding sourmash to the public CLI.

Merge now?

**Comment by kiepczi** [comment]
Sure. I will merge now and work on the next steps on the separate branch :)

**Comment by peterjc** [comment]
We missed adding ``tests/generate_fixtures/generate_target_sourmash_files.py`` to the ``Makefile`` for ``make fixtures`` - might as well do it on #174?

**Comment by kiepczi** [comment]
Not a problem, I will take care of this! Thanks for spotting this :)

**Pull request merged**